src/client/js/partials/exposure-scan.js

```diff
@@ -0,0 +1,109 @@
+  } else {
+    showNoBreachesResult(responseBody)
+  }
+}
+
+/**
+ * @param {import("../../../controllers/request-breach-scan").RequestBreachScanSuccessResponse} response
+ */
+function showOverflowingBreachResults (response) {
+  const heading = document.createElement('h1')
+  heading.innerHTML = response.heading
+  exposureScanPartial.querySelector('#exposure-scan-results-overflow .exposure-scan-hero-content').insertAdjacentElement('afterbegin', heading)
+  const breachCards = response.breaches.map((breach, index) => getBreachCard(breach, response.logos[index], response.dataClassStrings[index]))
+  const breachContainer = exposureScanPartial.querySelector('#exposure-scan-results-overflow .exposure-scan-breaches')
+  breachCards.forEach(card => breachContainer.appendChild(card))
+  exposureScanPartial.querySelector('#exposure-scan-loading').hidden = true
+  exposureScanPartial.querySelector('#exposure-scan-results-overflow').hidden = false
+}
+
+/**
+ * @param {import("../../../controllers/request-breach-scan").RequestBreachScanSuccessResponse} response
+ */
+function showSomeBreachResults (response) {
+  const heading = document.createElement('h1')
+  heading.innerHTML = response.heading
+  exposureScanPartial.querySelector('#exposure-scan-results-some .exposure-scan-hero-content').insertAdjacentElement('afterbegin', heading)
+  const breachCards = response.breaches.map((breach, index) => getBreachCard(breach, response.logos[index], response.dataClassStrings[index]))
+  const breachContainer = exposureScanPartial.querySelector('#exposure-scan-results-some .exposure-scan-breaches')
+  breachCards.forEach(card => breachContainer.appendChild(card))
+  exposureScanPartial.querySelector('#exposure-scan-loading').hidden = true
+  exposureScanPartial.querySelector('#exposure-scan-results-some').hidden = false
+}
+
+/**
+ * @param {import("../../../controllers/request-breach-scan").RequestBreachScanSuccessResponse} response
+ */
+function showNoBreachesResult (response) {
+  const heading = document.createElement('h1')
+  heading.innerHTML = response.heading
+  exposureScanPartial.querySelector('#exposure-scan-results-none .exposure-scan-hero-content').insertAdjacentElement('afterbegin', heading)
+  exposureScanPartial.querySelector('#exposure-scan-loading').hidden = true
+  exposureScanPartial.querySelector('#exposure-scan-results-none').hidden = false
+}
+
+function getBreachCard (breach, logo, dataClassStrings) {
+  const newCard = document.getElementById('breach-template').content.cloneNode(true)
+  newCard.querySelector('.exposure-scan-breach-company-logo').innerHTML = logo
+  newCard.querySelector('.exposure-scan-breach-company-name').textContent = breach.Title
+  newCard.querySelector('.exposure-scan-breach-added dd').textContent = new Date(breach.AddedDate).toLocaleString(navigator.languages, { year: 'numeric', month: 'long', day: 'numeric' })
+  newCard.querySelector('.exposure-scan-breach-data dd').textContent = formatList(dataClassStrings)
```

Comment:
Added comments to the empty elements in 01ec944c4.

Also happy to add more classes, but the class names had been getting so long already, and this one was already specific enough to be confident that it would not match other `<dd>` elements, that I figured it was probably less visual noise to avoid them.

(Underappreciated benefit of JSX: removing this layer of indirection.)